Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -6952,10 +6953,14 @@ static RefPtr<CSSRayValue> consumeRayShape(CSSParserTokenRange& range, const CSS
         return nullptr;
     }
 
-    // <angle> and <size> must be present.
-    if (!angle || !size)
+    // <angle> must be present.
+    if (!angle)
         return nullptr;
 
```

Comment:
Seems like `CSSRayValue` should store a `CSSValueID`, not a `CSSValue`, for `ray-size`. More efficient in memory use and also for almost every operation we would do with it, like serialization.

also, for parsing the `isContaining` boolean we should use `consumeIdentRaw` so we don’t create and then immediately destroy a `CSSPrimitiveValue` as part of parsing.